src/nools/task-defaults.js

```diff
@@ -0,0 +1,38 @@
+// File defining task defaults
+function taskDefaults(Utils) {
+  return {
+    defaultResolvedIf: function (contact, report, event, dueDate) {
+      var resolvingForm = getDefaultResolvingForm(this.definition);
+      if (typeof resolvingForm !== 'string') {
+        throw new Error('Could not find the default resolving form!');
+      }
+      var start = 0;
+      if (report) {//Report based task
+        //Getting the latest date between start of the task's window and just after the report's reported date.
+        start = Math.max(Utils.addDate(dueDate, -event.start).getTime(), report.reported_date + 1);
+      }
+      else {
+        start = Utils.addDate(dueDate, -event.start).getTime();
+      }
+      var end = Utils.addDate(dueDate, event.end + 1).getTime();
+      return Utils.isFormSubmittedInWindow(
+        contact.reports,
+        resolvingForm,
+        start,
+        end
+      );
+    }
+  };
+}
+
+function getDefaultResolvingForm(taskDefinition) {
+  var resolvingAction = taskDefinition && taskDefinition.actions &&
```

Comment:
You can exit early, it'll require less effort to understand the conditions and will keep lines shorter. Specially if we're calling a function with a callback, in that case better assign the returning value to a variable and then use the variable in the condition.
Usually we treat falsy values ("", null, undefined, false, 0) the same, so we can avoid using "typeof" which is for specific cases where we need to differentiate from any of those values, for example. (Is it a specific case?)

```
function getDefaultResolvingForm(taskDefinition) {
  var resolvingAction;    // --> because of using "var", better here to avoid hoisting

  if (!taskDefinition || !taskDefinition.actions) { // --> exit early
    return;
  }

  resolvingAction = taskDefinition.actions.find(
      function (action) {
        return !action.type || action.type === 'report' // --> no typeof
      }
    );

  return resolvingAction && resolvingAction.form;
}
```